Mimir/Controllers/TableSheetsController.cs

```diff
@@ -12,23 +9,38 @@ namespace Mimir.Controllers;
 public class TableSheetsController(TableSheetsRepository tableSheetsRepository) : ControllerBase
 {
     [HttpGet("names")]
-    public async Task<string[]> GetSheetNames(
-        string network
-    )
+    public async Task<string[]> GetSheetNames(string network)
     {
         var sheetNames = tableSheetsRepository.GetSheetNames(network);
 
         return sheetNames;
     }
 
     [HttpGet("{sheetName}")]
-    public async Task<ContentResult> GetSheet(
-        string network,
-        string sheetName
-    )
+    [Produces("application/json", "text/csv")]
+    public async Task<IActionResult> GetSheet(string network, string sheetName)
     {
-        var sheet = tableSheetsRepository.GetSheet(network, sheetName);
+        var acceptHeader = Request.Headers["Accept"].ToString();
 
-        return Content(sheet, "application/json");;
+        SheetFormat sheetFormat = acceptHeader.Contains("text/csv")
+            ? SheetFormat.Csv
+            : SheetFormat.Json;
```

Comment:
It should return NotAcceptable, 406 response if it isn't acceptable.

```
        SheetFormat? sheetFormatNullable = acceptHeader switch
        {
            "text/csv" => SheetFormat.Csv,
            "application/json" => SheetFormat.Json,
            _ => null,
        };

        if (sheetFormatNullable is not { } sheetFormat)
        {
            return StatusCode(StatusCodes.Status406NotAcceptable);
        }
```

Or it should implement `NotAcceptibleAcception` and exception filter feature.

ref:
- https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406

Reply:
Good status code, applied this code